components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsResourceImplTest.java

```diff
@@ -0,0 +1,226 @@
+        verify(this.logger).debug("Importing new patient from JSON via REST: {}", (String) null);
+    }
+
+    @Test
+    public void creatingPatientFails() {
+        JSONObject json = new JSONObject();
+        Exception exception = new NullPointerException();
+        doReturn(true).when(this.access).hasAccess(eq(Right.EDIT), any(DocumentReference.class), any(EntityReference.class));
+        doThrow(exception).when(this.repository).createNewPatient();
+        Response response = this.patientsResource.addPatient(json.toString());
+        Assert.assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), response.getStatus());
+        verify(this.logger).error("Could not process remote matching request: {}", exception.getMessage(), exception);
+    }
+
+    @Test
+    public void addPatientAsJSON() {
+        doReturn(true).when(this.access).hasAccess(eq(Right.EDIT), any(DocumentReference.class), any(EntityReference.class));
+        doReturn(this.patient).when(this.repository).createNewPatient();
+        JSONObject jsonPatient = new JSONObject();
+        Response response = this.patientsResource.addPatient(jsonPatient.toString());
+        Assert.assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+        verify(this.logger).debug("Importing new patient from JSON via REST: {}", jsonPatient.toString());
+    }
+
+    @Test
+    public void listPatientsNullOrderField() throws QueryException {
+        Query query = mock(DefaultQuery.class);
+        doReturn(query).when(this.queries).createQuery(anyString(), anyString());
+        doReturn(query).when(query).bindValue(anyString(), anyString());
+        doReturn(new ArrayList<Object[]>()).when(query).execute();
+        Patients result = this.patientsResource.listPatients(0, 30, null, "asc");
+        verify(this.queries).createQuery("select doc.fullName, p.external_id, doc.creator, doc.creationDate, doc.version, doc.author, doc.date"
+                + " from Document doc, doc.object(PhenoTips.PatientClass) p where doc.name <> :t order by "
+                + "doc.name" + " asc", "xwql");
+    }
+
+    @Test
+    public void listPatientsNullOrder() throws QueryException {
+        Query query = mock(DefaultQuery.class);
+        doReturn(query).when(this.queries).createQuery(anyString(), anyString());
+        doReturn(query).when(query).bindValue(anyString(), anyString());
+        doReturn(new ArrayList<Object[]>()).when(query).execute();
+        Patients result = this.patientsResource.listPatients(0, 30, "id", null);
+        verify(this.queries).createQuery("select doc.fullName, p.external_id, doc.creator, doc.creationDate, doc.version, doc.author, doc.date"
+                + " from Document doc, doc.object(PhenoTips.PatientClass) p where doc.name <> :t order by "
+                + "doc.name" + " asc", "xwql");
+    }
+
+    @Test
+    public void listPatientsNonDefaultBehaviour() throws WebApplicationException, QueryException {
```

Comment:
`WebApplicationException` is a `RuntimeException` you don't need a `throws` clause for it